consumption.tex

```diff
@@ -0,0 +1,50 @@
+
+Knowing this, it is now possible to define \textsf{blockConnection}. This component will be an MMR proof for the fact that that $id(\mathsf{blockHeader}) \in \text{MMR}$ in the form of $\mathsf{blockConnection} = [(x_0, s_0), (x_1, s_1), \dots]$, where $s_i \in \sf\{Left, Right\}$.
+
+An interesting observation is that, because the federation majority is trusted on the target blockchain, we could avoid doing the block header hashing and verification in order to verify block connection. Instead of storing the block id of each block in the MMR, we could store the transaction merkle root of each block, and again use the root of the MMR as the tag. The verifier on the smart contract side would then verify that the \textsf{txIDMerkleRoot} provided exists in the MMR using the \textsf{blockConnection}.
+
+\import{./}{algorithms/alg.verify-event-federation.tex}
+
+\subsubsection{Decentralized}
+We introduce a second way for proving and verifying block connection which does not require a trusted federation. This is possible by making use of NIPoPoWs. The way the proving process works differs quite a bit from the federation, because in the federation there is a trusted source of truth. If a tag is voted by the federation majority, then it is forever true. But if this the verifier does not have this information, then it has to rely on the prover to get it. We will need to assume a model where there exist two provers, one honest and one malicious. The block connection proof can be an NIPoPoW infix proof for the block which includes the burn transaction. Specifically, $\mathsf{blockConnection} = \pi\chi$.
+
+\import{./}{algorithms/alg.verify-event-nipopow.tex}
+
+However, this alone is not enough to persuade a verifier. It is possible that the prover is malicious and this proof is constructed for a non-honestly adopted chain (e.g. a temporary adversarial fork). In order for the verifier to be certain, it needs to allow for a honest prover to claim a better proof for a longer chain. 
+
+\subsection{Smart Contract Implementation}
+\import{./}{algorithms/alg.burn-verifier.tex}
+
+We present a smart contract in Algorithm~\ref{alg.burn-verifier} which we call the \textsf{burn-verifier}. The \textsf{burn-verifier} verifies burn events from the source chain. The contract extends the \textsf{crosschain} contract from \cite{pow-sidechains}. The \textsf{crosschain} contract is responsible for registering events on the source blockchain and verifying them. For our application, we parameterise \textsf{crosschain} with a \textsf{verify} function. The function presented in Algorithm~\ref{alg.verify-event-federation} will be used in a federated setup, and the function presented in Algorithm~\ref{alg.verify-event-nipopow} will be used in a decentralized setup.
```

Comment:
I feel we need a more detailed introduction to the "PoW sidechains" work here; the expert reader will not be familiar with this work.

Reply:
parametrise